nf_core/create.py

```diff
@@ -525,19 +527,22 @@ def git_init_pipeline(self):
         """
         # Check that the default branch is not dev
         try:
-            default_branch = git.config.GitConfigParser().get_value("init", "defaultBranch")
+            default_branch = self.default_branch or git.config.GitConfigParser().get_value("init", "defaultBranch")
         except configparser.Error:
             default_branch = None
             log.debug("Could not read init.defaultBranch")
-        if default_branch == "dev" or default_branch == "TEMPLATE":
+        if default_branch in ["dev", "TEMPLATE"]:
             raise UserWarning(
-                f"Your Git defaultBranch is set to '{default_branch}', which is incompatible with nf-core.\n"
-                "This can be modified with the command [white on grey23] git config --global init.defaultBranch <NAME> [/]\n"
-                "Pipeline git repository is not initialised."
+                f"Your Git defaultBranch '{default_branch}' is incompatible with nf-core.\n"
+                "Set the default branch name with "
+                "[white on grey23] git config --global init.defaultBranch <NAME> [/]\n"
+                # Or set the default_branch parameter in this class
+                "Pipeline git repository will not be initialised."
             )
         # Initialise pipeline
         log.info("Initialising pipeline git repository")
-        repo = git.Repo.init(self.outdir)
+        init_kwargs = {"initial_branch": default_branch} if default_branch is not None else {}
+        repo = git.Repo.init(self.outdir, **init_kwargs)
```

Comment:
This will fail on machines with `git < 2.28`, e.g. Ubuntu 20.04 still uses git 2.25. Maybe catch the error and either raise a user warning or add a line to rename the default branch?

Reply:
good catch. Maybe even better would be to conditional renaming to the default branch to the supplied default branch name if one is set. I think this makes it easier to read in addition to the added robustness against git version changes.